packages/grid/x-data-grid-pro/src/tests/editRows.DataGridPro.test.tsx

```diff
@@ -1520,6 +1520,34 @@ describe('<DataGridPro /> - Edit Rows', () => {
         expect(secondInput).to.have.attribute('aria-invalid', 'true');
       });
     });
+
+    it("should exit the row edit mode and save the row when preProcessEditCellProps don't return an error", async () => {
+      render(
+        <TestCase
+          editMode="row"
+          columns={[
+            {
+              field: 'brand',
+              editable: true,
+              preProcessEditCellProps: ({ props }) => props,
+            },
+          ]}
+          rows={[{ id: 0, brand: 'Nike' }]}
+        />,
+      );
+      const cell = getCell(0, 0);
+      cell.focus();
```

Comment:
The internal state is not updated when only `cell.focus()` is called. See #3486.

```suggestion
      fireEvent.mouseUp(cell);
```

Reply:
Other tests should probably updated then, I copied this logic :+1:

Reply:
Fixed + TODO for other tests